index.js

```diff
@@ -11,8 +11,8 @@ const VERSION = 3; // serialized format version
 export default class Flatbush {
 
     static from(data) {
-        if (!(data instanceof ArrayBuffer)) {
-            throw new Error('Data must be an instance of ArrayBuffer.');
+        if (!(data instanceof ArrayBuffer) && SharedArrayBuffer && !(data instanceof SharedArrayBuffer)) {
```

Comment:
Two issues here, if I'm following this right:

- In case `SharedArrayBuffer` isn't available, just referencing it will throw an error, so this won't work as a check. We need to use `typeof`.
- If we fix the reference, the whole condition will still evaluate to `false` in case of a wrong data type, so the error won't be thrown.

So this would need to look something like this:

```suggestion
        if (!(data instanceof ArrayBuffer) && (typeof SharedArrayBuffer === 'undefined' || !(data instanceof SharedArrayBuffer))) {
```